src/main/java/ru/hogwarts/school/repository/StudentRepository.java

```diff
@@ -1,8 +1,25 @@
 package ru.hogwarts.school.repository;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 import ru.hogwarts.school.model.Student;
+
+import java.util.List;
+import java.util.Optional;
+
 @Repository
-public interface StudentRepository extends JpaRepository<Student,Long> {
+public interface StudentRepository extends JpaRepository<Student, Long> {
+    List<Student> findByAgeBetween(int startAge, int endAge);
+
+    Optional<Student> findById(long id);
+
+    @Query(value = "SELECT id, SUM(id) as student FROM students GROUP BY name", nativeQuery = true)
```

Comment:
Не то получаешь - у тебя считается сумма, а надо количеством

```suggestion
    @Query(value = "select count(*) from student", nativeQuery = true)
```